lib/lookup/src/lookup_v2/mod.rs

```diff
@@ -22,12 +22,42 @@ impl<'de> Deserialize<'de> for OwnedPath {
 }
 
 impl Serialize for OwnedPath {
-    fn serialize<S>(&self, _serializer: S) -> Result<S::Ok, S::Error>
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: Serializer,
     {
-        // TODO: Implement canonical way to serialize segments.
-        todo!()
+        if self.segments.is_empty() {
+            serializer.serialize_str(".")
+        } else {
+            let path = self
+                .segments
+                .iter()
+                .map(|segment| match segment {
+                    OwnedSegment::Field(field) => {
+                        let needs_quotes = field
+                            .chars()
+                            .any(|c| !matches!(c, 'A'..='Z' | 'a'..='z' | '_' | '0'..='9' | '@'));
+                        if needs_quotes {
+                            let mut string = String::new();
```

Comment:
why not `String::with_capacity(field.as_bytes().len())`?

Reply:
Also just noticed this when reviewing my own code 😄 That was a clunky leftofer from some refactoring. Went with a slightly different approach in a8d18d2450bff6db7bf0a010af68419467d2c347, but same idea

Reply:
5a06ba592bd783dd26baabb1f083e2973a828770 looks better now and is also what you suggested 😄